src/storage-client/src/controller.rs

```diff
@@ -2262,6 +2298,14 @@ where
                 }
             }
 
+            if frontier.is_empty() {
+                if self.state.collections.get(&id).is_some() {
```

Comment:
Double removes are fine, so why do we need to check that the collection still exists?

Reply:
sink's and source's always have non-overlapping id'?

Reply:
Ah, just being conservative here--makes sense!